gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14531,4 +14532,8 @@ private void memTrace(String mesage) {
     }
   }
 
+  public boolean isInternalColumnTable() {
+    return this.getName().toUpperCase().endsWith(StoreCallbacks.SHADOW_TABLE_SUFFIX);
+  }
```

Comment:
@rishitesh It is preferable to make this a final boolean field in LocalRegion instead of the method which is expensive (change to upper-case then check). Also as mentioned in the comment, I had looked through the code and there were many other similar checks. So all those uses (check the references to SHADOW_TABLE_SUFFIX) should use this method.